framework/PostProcessors/Metric.py

```diff
@@ -49,16 +49,17 @@ class cls.
       @ Out, inputSpecification, InputData.ParameterInput, class to use for
         specifying input of cls.
     """
-    ## This will replace the lines above
-    inputSpecification = super(Metrics, cls).getInputSpecification()
-
-    ## TODO: Fill this in with the appropriate tags
-    #FeaturesInput = InputData.parameterInputFactory("Features", contentType=InputData.StringType)
-    #inputSpecification.addSub(FeaturesInput)
-    #TargetsInput = InputData.parameterInputFactory("Targets", contentType=InputData.StringType)
-    #inputSpecification.addSub(TargetsInput)
-    #MetricInput = InputData.parameterInputFactory("Metric", contentType=InputData.StringType)
-    #inputSpecification.addSub(MetricInput)
+    inputSpecification = super(Metric, cls).getInputSpecification()
+    FeaturesInput = InputData.parameterInputFactory("Features", contentType=InputData.StringType)
+    FeaturesInput.addParam("type", InputData.StringType)
+    inputSpecification.addSub(FeaturesInput)
+    TargetsInput = InputData.parameterInputFactory("Targets", contentType=InputData.StringType)
+    TargetsInput.addParam("type", InputData.StringType)
+    inputSpecification.addSub(TargetsInput)
+    MetricInput = InputData.parameterInputFactory("Metric", contentType=InputData.StringType)
+    MetricInput.addParam("class", InputData.StringType)
```

Comment:
since it is an error if these are not there, it probably should be:
`MetricInput.addParam("class", InputData.StringType, True)`
which makes it required.  Ditto for `type`.

Reply:
Got it, thanks.